### PR TITLE
Editor: hide published/updated popup in 2 seconds

### PR DESCRIPTION
**Pull request by ShreyaSingh10**
Fix for https://github.com/Automattic/wp-calypso/issues/19354

### CODE REVIEW

File: client/post-editor/editor-notice/index.jsx [hunk 1]
```diff
@@ -292,6 +292,21 @@ export class EditorNotice extends Component {
 		const { siteId, message, status, onDismissClick } = this.props;
 		const text = this.getErrorMessage() || this.getText( message );
 
+		if ( message === 'updated' || message === 'published' ) {
+			return (
+				<div className={ classNames( 'editor-notice', { 'is-global': true } ) }>
```
Comment (alisterscott) [inline client/post-editor/editor-notice/index.jsx]: This looks like it duplicates the code below it

File: client/post-editor/editor-notice/index.jsx [hunk 2]
```diff
@@ -292,6 +292,21 @@ export class EditorNotice extends Component {
...
+					{ siteId && <QueryPostTypes siteId={ siteId } /> }
+					{ text && (
+						<Notice
+							{ ...{ status, text, onDismissClick } }
+							duration={ 2000 }
```
Comment (alisterscott) [inline client/post-editor/editor-notice/index.jsx]: could you add this as a conditional to the existing block of code?

Comment (ShreyaSingh10) [reply client/post-editor/editor-notice/index.jsx]: Do you think I should move the parameter to render call or to the parent template?

File: client/post-editor/editor-notice/index.jsx (hunk 1 above)
Comment (ShreyaSingh10) [reply client/post-editor/editor-notice/index.jsx]: Can you elaborate how do I separate this?

**Comment by ShreyaSingh10** [comment]
@alisterscott I have made the requested changes, let me know if that works.

Thank you!

**Comment by alisterscott** [comment]
Thanks for the update @ShreyaSingh10 and sorry for the delay in response. I will have a look at the update in the next day and let you know. Thanks

**Comment by vindl** [comment]
Thanks for contributing @ShreyaSingh10! Your new changes look good and address the code duplication issue that @alisterscott mentioned. We could simplify this further by always rendering the same `Notice` and passing different values for `duration` prop to it, depending on the value of `message`. For example:

```javascript
// duration prop defaults to 0 when it's not supplied
const noticeDuration = ( message === 'updated' || message === 'published' ) ? 2000 : 0;
...
// in render()
{ text && <Notice { ...{ status, text, onDismissClick } } duration={ noticeDuration } showDismiss={ true } /> }
```

However, while testing I noticed that there other message types for which the above issue still persists. For example it's still reproducible when scheduling a post. A couple of possible approaches to fixing this that come to my mind are:

1. Always provide some fixed duration for every notice (might not be suitable for errors).
2. Audit the code in https://github.com/Automattic/wp-calypso/blob/master/client/post-editor/post-editor.jsx and add every success notice type to the `noticeDuration` check that we perform above (could get out of sync and resurface the issue when someone adds a new notice message in editor and forgets to update it here).
3. Try to detect view changes in https://github.com/Automattic/wp-calypso/blob/master/client/post-editor/post-editor.jsx and utilize the existing `hideNotice()` method there to hide it.

The third approach looks like the best way to go, but let me know what do you all think.

**Comment by Copons** [comment]
Hi @ShreyaSingh10! Thanks for this PR, it takes care of something that has been bugging me for quite some time now! 🙂 

I think your approach in itself is valid, but I have to agree with @vindl.
His 3rd suggestion should be in fact the way to go.

Ideally, when we click on the Edit button, we would like to remove the notice altogether.
That button has a convenient `onEdit` prop that gets called on click.
If you backtrack that `onEdit` prop across a multitude of files, you'll eventually get to `post-editor.jsx`, with its [`this.onPreviewEdit()` method](https://github.com/Automattic/wp-calypso/blob/master/client/post-editor/post-editor.jsx#L813).

Looking at this file, you'll see that it also handles the `<EditorNotice>` rendering, thanks to the component internal state: `this.state.notice`.
As @vindl pointed out, here's also a `this.hideNotice()` method that [does what we want](https://github.com/Automattic/wp-calypso/blob/master/client/post-editor/post-editor.jsx#L228-L230): `this.setState( { notice: null } );`.

Though, in `this.onPreviewEdit()` we are [already setting the component internal state](https://github.com/Automattic/wp-calypso/blob/master/client/post-editor/post-editor.jsx#L823-L827).

I think by just "nulling" the notice in that method is enough for our needs:
```es6
this.setState( {
	showPreview: false,
	isPostPublishPreview: false,
	previewAction: null,
	notice: null,
} );
```

(This would also take care of the same issue happening upon post scheduling)

Do you think this makes sense?
Please feel free to ping me if you need any clarifications! 🙂

**Comment by apeatling** [comment]
@ShreyaSingh10 Closing this because of inactivity, if you pick it back up to make the changes, please reopen.

**Pull request closed without merging**